docs/architecture.md

```diff
@@ -15,139 +15,207 @@ Prometheus instances scrape samples from various targets and then push them to C
+
+## Storage
+
+Cortex currently supports two storage engines to store and query the time series:
+
+- Chunks (default, stable)
+- Blocks (experimental)
+
+The two engines mostly share the same Cortex architecture with few differences outlined in the rest of the document.
+
+### Chunks storage (default)
+
+The chunks storage stores each single time series into a separate object called chunk. Each chunk contains the timestamp-value pairs for a given period (defaults to 12 hours) of the series. Series chunks are then indexed by time range period and metric labels, in order to provide a fast lookup across millions - or even billions - of chunks.
+
+For this reason, the chunks storage consists of:
+
+* An index for the chunks. This index can be backed by:
+    * [Amazon DynamoDB](https://aws.amazon.com/dynamodb)
+    * [Google Bigtable](https://cloud.google.com/bigtable)
+    * [Apache Cassandra](https://cassandra.apache.org)
+* An object store for the chunk data itself, which can be:
+    * [Amazon DynamoDB](https://aws.amazon.com/dynamodb)
+    * [Google Bigtable](https://cloud.google.com/bigtable)
+    * [Apache Cassandra](https://cassandra.apache.org)
+    * [Amazon S3](https://aws.amazon.com/s3)
+    * [Google Cloud Storage](https://cloud.google.com/storage/)
+
+Internally, the access to the chunks storage relies on a unified interface called "chunks store". Unlike other Cortex components, the chunk store is not a separate service, but rather a library embedded in the services that need to access the long-term storage: [ingester](#ingester), [querier](#querier) and [ruler](#ruler).
+
+The chunk and index data format is versioned, in order to allow Cortex operators to upgrade the cluster to the latest stable version and take advantage of new features and improvements - which may introduce changes in the storage format - without requiring any downtime or complex procedures to rewrite the stored data.
+
+A set of storage schemas are used to map the version to use to read and write time series belonging to a specific period of time. The current schema recommendation is the **v10 schema** (v11 is still experimental). For more information about the schema, please check out the "[Schema](guides/running.md#schema)" documentation.
+
+### Blocks storage (experimental)
+
+The blocks storage is based on [Prometheus TSDB](https://prometheus.io/docs/prometheus/latest/storage/): it stores all the time series for a given tenant and period (defaults to 2 hours) into a single TSDB block. Each block is composed by few files storing the chunks and the block index.
+
+The chunk files contain the timestamp-value pairs for multiple series. The series inside the chunks are then indexed by a per-block index, which indexes metric names and labels to time series in the chunk files.
```

Comment:
When we talk about chunks, are we referring to the same chunks previously defined or TSDB chunks? Would be good to clear up that distinction and link to https://github.com/prometheus/prometheus/blob/master/tsdb/docs/format/chunks.md if necessary.

Reply:
The chunks in the blocks storage are TSDB chunks. Conceptually they're not much different from the "chunks storage" given they both contain timestamp-value pairs, but in TSDB a single chunks file contain multiple series (many).

I've changed it to "TSDB chunk files" without repeating "TSDB chunk" in every single place, cause looks a bit overkilled to me.